src/Shared/RunnerUtilities.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Microsoft.Build.Shared;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.SharedUtilities
+{
+
+    internal static class RunnerUtilities
+    {
+        public static string PathToMsBuildExe => FileUtilities.CurrentExecutablePath;
+
+        /// <summary>
+        /// Invoke msbuild.exe with the given parameters and return the stdout, stderr, and process exit status.
+        /// This method may invoke msbuild via other runtimes.
+        /// </summary>
+        public static string ExecMSBuild(string msbuildParameters, out bool successfulExit)
+        {
+            return ExecMSBuild(PathToMsBuildExe, msbuildParameters, out successfulExit);
+        }
+
+        /// <summary>
+        /// Invoke msbuild.exe with the given parameters and return the stdout, stderr, and process exit status.
+        /// This method may invoke msbuild via other runtimes.
+        /// </summary>
+        public static string ExecMSBuild(string pathToMsBuildExe, string msbuildParameters, out bool successfulExit)
+        {
+#if FEATURE_RUN_EXE_IN_TESTS
```

Comment:
I lean toward "yes" as well, now that it's pointed out.  YAGNI, and we can always move it to the product side later if necessary.